### PR TITLE
(660) Prison adjudications screen

### PR DESCRIPTION
**Pull request by pezholio**
This adds the Prison Adjudications section to the Case Notes. There's been a bit of refactoring going on in the CaseNotes class (open question as to whether this is even the right name now!), and a bit of frontend JS and CSS too. Lemme know if anything needs breaking down any further!

## Screenshot

image

### CODE REVIEW

**Comment by pezholio** [comment]
[Prison adjudications screen](https://trello.com/c/42fCCnfb/660-prison-adjudications-screen)

File: server/testutils/jest.setup.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+declare global {
+  namespace jest {
+    interface Matchers<R> {
+      toMatchStringIgnoringWhitespace(expected: string): R
+    }
+  }
+}
+
+expect.extend({
```
Comment (richpjames) [inline server/testutils/jest.setup.ts:12]: 👏👏👏

File: integration_tests/mockApis/person.ts [hunk 2]
```diff
@@ -59,4 +59,17 @@ export default {
         jsonBody: args.prisonCaseNotes,
       },
     }),
+
+  stubAdjudications: async (args: { person: Person; adjudications: Adjudication[] }) =>
```
Comment (richpjames) [inline integration_tests/mockApis/person.ts:63]: I don't *think* this needs to be `async` - same for `stubPrisonCaseNotes`

**Review by richpjames — APPROVED** [review]
Looks good, I have left a few comments. 
I wonder if just 'Prison Information' should replace Case Notes?

File: server/testutils/factories/adjudication.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import { faker } from '@faker-js/faker/locale/en_GB'
+
+import { DateFormats } from '../../utils/dateUtils'
+
+export default Factory.define<Adjudication>(() => ({
+  id: faker.datatype.number(),
+  reportedAt: DateFormats.formatApiDate(faker.date.past()),
+  establishment: faker.address.cityName(),
+  offenceDescription: faker.lorem.sentence(),
+  hearingHeld: faker.helpers.arrayElement([true, false]),
```
[the remark this answers is not in the document]
Comment (pezholio) [reply server/testutils/factories/adjudication.ts]: Ah, course 🤦

File: server/form-pages/apply/prison-information/caseNotes.test.ts [hunk 4]
```diff
@@ -43,13 +43,45 @@ describe('CaseNotes', () => {
       id: 'C',
       createdAt: DateFormats.formatApiDate(new Date(2022, 1, 0)),
       occurredAt: DateFormats.formatApiDate(new Date(2022, 0, 1)),
+describe('caseNoteCheckbox', () => {
```
[the remark this answers is not in the document]
Comment (pezholio) [reply server/form-pages/apply/prison-information/caseNotes.test.ts:61]: Yeah, that's an artefact of a bad merge 😬  Fixed now!

**Pull request merged**